core/src/main/java/org/ehcache/spi/lifecycle/LifeCycleListener.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.spi.lifecycle;
+
+/**
+ * Listens of life cycle change on a particular object
+ * 
+ * @author Mathieu Carbou
+ */
+public interface LifeCycleListener<T> {
+
+  /**
+   * Called when the instance is initialized
+   */
+  void onInitialized(T instance);
+
+  /**
+   * Called when the instance is closed
+   */
+  void onClosed(T instance);
```

Comment:
I feel `afterClosing(T instance)` would better indicate when you are called exactly.